lib/src/elastic_client_impl.dart

```diff
@@ -265,18 +270,70 @@ class Client {
       results,
       suggestHits: suggestHits.isEmpty ? null : suggestHits,
       aggregations: aggResult.isEmpty ? null : aggResult,
+      scrollId: body['_scroll_id'] as String ?? null,
+    );
+  }
+
+  Future<SearchResult> scroll(String scrollId, Duration scroll) async {
```

Comment:
I think it would be nice if this would be part of the `SearchResult` object, e.g. `SearchResult.nextPage()`. But it is probably outside of the scope of the current PR, I can probably do a generic API overhaul later...

Reply:
The abstraction is great, but I think it's nice to keep it close to the REST API.

The Java client looks like this.
https://www.elastic.co/guide/en/elasticsearch/client/java-rest/master/java-rest-high-search-scroll.html#_retrieve_all_the_relevant_documents